<?xml version="1.0" encoding="UTF-8"?>
<!-- Generated by org.testng.reporters.JUnitReportReporter -->
<testsuite hostname="Shweta-PC" failures="13" tests="123" name="dsAlgo_TestClasses.Tree_TestClass" time="1085.587" errors="1" timestamp="2025-03-13T18:05:09 EDT" skipped="28">
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="callValidLogInData" time="7.485"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="validLoginDataBase" time="4.621"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="OverviewOfTrees" time="10.796"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="OverviewOfTreesTryhere" time="11.894"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInOverviewOfTreesPage" time="8.229">
    <failure message="Test failed: Expected an error message when clicking the &#039;Run&#039; button without entering code, but no message was displayed" type="java.lang.AssertionError">
      <![CDATA[java.lang.AssertionError: Test failed: Expected an error message when clicking the 'Run' button without entering code, but no message was displayed
at org.testng.Assert.fail(Assert.java:111)
at dsAlgo_TestClasses.Tree_TestClass.checkErrorMessageWhenExecutingWithoutenteringCodeInOverviewOfTreesPage(Tree_TestClass.java:67)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:269)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:76)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
]]>
    </failure>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInOverviewOfTreesPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInOverviewOfTreesPage" time="8.376">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInOverviewOfTreesPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInOverviewOfTreesPage" time="9.009">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInOverviewOfTreesPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="OverviewOfTreesTryeditor" time="12.907"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="OverviewOfTreesTryeditor" time="13.563"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="OverviewOfTreesTryeditor" time="13.562"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="navigateback" time="8.439"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="Terminologies" time="10.156"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="TerminologiesTryhere" time="8.707"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInTerminologiesPage" time="8.722">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInTerminologiesPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInTerminologiesPage" time="8.354">
    <failure message="Test failed: Expected an error message when clicking the &#039;Run&#039; button without entering code, but no message was displayed" type="java.lang.AssertionError">
      <![CDATA[java.lang.AssertionError: Test failed: Expected an error message when clicking the 'Run' button without entering code, but no message was displayed
at org.testng.Assert.fail(Assert.java:111)
at dsAlgo_TestClasses.Tree_TestClass.checkErrorMessageWhenExecutingWithoutenteringCodeInTerminologiesPage(Tree_TestClass.java:115)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:269)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:76)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
]]>
    </failure>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInTerminologiesPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInTerminologiesPage" time="8.466">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInTerminologiesPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="TerminologiesTryeditor" time="8.993"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="TerminologiesTryeditor" time="9.176"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="TerminologiesTryeditor" time="9.076"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="navigatebacktoTerminologies" time="8.669"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="TypesOfTrees" time="7.666"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="TypesOfTreesTryhere" time="8.746"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInTypesOfTreesPage" time="8.728">
    <failure message="Test failed: Expected an error message when clicking the &#039;Run&#039; button without entering code, but no message was displayed" type="java.lang.AssertionError">
      <![CDATA[java.lang.AssertionError: Test failed: Expected an error message when clicking the 'Run' button without entering code, but no message was displayed
at org.testng.Assert.fail(Assert.java:111)
at dsAlgo_TestClasses.Tree_TestClass.checkErrorMessageWhenExecutingWithoutenteringCodeInTypesOfTreesPage(Tree_TestClass.java:161)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:269)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:76)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
]]>
    </failure>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInTypesOfTreesPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInTypesOfTreesPage" time="8.156">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInTypesOfTreesPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInTypesOfTreesPage" time="7.954">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInTypesOfTreesPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="TypesOfTreesTryeditor" time="8.979"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="TypesOfTreesTryeditor" time="8.846"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="TypesOfTreesTryeditor" time="9.230"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="navigatebacktoTypesOfTrees" time="8.608"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="TreeTraversals" time="7.867"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="TreeTraversalsTryhere" time="8.283"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInTreeTraversalsPage" time="8.853">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInTreeTraversalsPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInTreeTraversalsPage" time="8.566">
    <failure message="Test failed: Expected an error message when clicking the &#039;Run&#039; button without entering code, but no message was displayed" type="java.lang.AssertionError">
      <![CDATA[java.lang.AssertionError: Test failed: Expected an error message when clicking the 'Run' button without entering code, but no message was displayed
at org.testng.Assert.fail(Assert.java:111)
at dsAlgo_TestClasses.Tree_TestClass.checkErrorMessageWhenExecutingWithoutenteringCodeInTreeTraversalsPage(Tree_TestClass.java:206)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:269)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:76)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
]]>
    </failure>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInTreeTraversalsPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInTreeTraversalsPage" time="7.845">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInTreeTraversalsPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="TreeTraversalsTryeditor" time="9.223"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="TreeTraversalsTryeditor" time="8.202"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="TreeTraversalsTryeditor" time="9.890"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="navigatebacktoTreeTraversals" time="7.925"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="TraversalsIllustration" time="7.681"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="TraversalsIllustrationTryhere" time="7.691"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInTraversalsIllustrationPage" time="8.348">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInTraversalsIllustrationPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInTraversalsIllustrationPage" time="8.943">
    <failure message="Test failed: Expected an error message when clicking the &#039;Run&#039; button without entering code, but no message was displayed" type="java.lang.AssertionError">
      <![CDATA[java.lang.AssertionError: Test failed: Expected an error message when clicking the 'Run' button without entering code, but no message was displayed
at org.testng.Assert.fail(Assert.java:111)
at dsAlgo_TestClasses.Tree_TestClass.checkErrorMessageWhenExecutingWithoutenteringCodeInTraversalsIllustrationPage(Tree_TestClass.java:252)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:269)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:76)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
]]>
    </failure>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInTraversalsIllustrationPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInTraversalsIllustrationPage" time="8.201">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInTraversalsIllustrationPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="TraversalsIllustrationTryeditor" time="8.870"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="TraversalsIllustrationTryeditor" time="9.155"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="TraversalsIllustrationTryeditor" time="9.313"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="navigatebacktoTraversalsIllustration" time="8.689"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="BinaryTrees" time="8.077"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="BinaryTreesTryhere" time="8.188"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInBinaryTreesPage" time="8.562">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInBinaryTreesPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInBinaryTreesPage" time="8.122">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInBinaryTreesPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInBinaryTreesPage" time="8.770">
    <failure message="Test failed: Expected an error message when clicking the &#039;Run&#039; button without entering code, but no message was displayed" type="java.lang.AssertionError">
      <![CDATA[java.lang.AssertionError: Test failed: Expected an error message when clicking the 'Run' button without entering code, but no message was displayed
at org.testng.Assert.fail(Assert.java:111)
at dsAlgo_TestClasses.Tree_TestClass.checkErrorMessageWhenExecutingWithoutenteringCodeInBinaryTreesPage(Tree_TestClass.java:299)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:269)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:76)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
]]>
    </failure>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInBinaryTreesPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="BinaryTreesTryeditor" time="9.267"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="BinaryTreesTryeditor" time="9.320"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="BinaryTreesTryeditor" time="9.617"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="navigatebacktoBinaryTrees" time="8.163"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="TypesOfBinaryTrees" time="8.188"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="TypesOfBinaryTreesTryhere" time="8.733"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInTypesOfBinaryTreesPage" time="8.058">
    <failure message="Test failed: Expected an error message when clicking the &#039;Run&#039; button without entering code, but no message was displayed" type="java.lang.AssertionError">
      <![CDATA[java.lang.AssertionError: Test failed: Expected an error message when clicking the 'Run' button without entering code, but no message was displayed
at org.testng.Assert.fail(Assert.java:111)
at dsAlgo_TestClasses.Tree_TestClass.checkErrorMessageWhenExecutingWithoutenteringCodeInTypesOfBinaryTreesPage(Tree_TestClass.java:346)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:269)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:76)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
]]>
    </failure>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInTypesOfBinaryTreesPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInTypesOfBinaryTreesPage" time="8.295">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInTypesOfBinaryTreesPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInTypesOfBinaryTreesPage" time="8.289">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInTypesOfBinaryTreesPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="TypesOfBinaryTreesTryeditor" time="8.898"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="TypesOfBinaryTreesTryeditor" time="9.286"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="TypesOfBinaryTreesTryeditor" time="8.630"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="navigatebacktoTypesOfBinaryTrees" time="8.793"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="ImplementationInPython" time="7.984"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="ImplementationInPythonTryhere" time="8.792"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInImplementationInPythonPage" time="8.196">
    <failure message="Test failed: Expected an error message when clicking the &#039;Run&#039; button without entering code, but no message was displayed" type="java.lang.AssertionError">
      <![CDATA[java.lang.AssertionError: Test failed: Expected an error message when clicking the 'Run' button without entering code, but no message was displayed
at org.testng.Assert.fail(Assert.java:111)
at dsAlgo_TestClasses.Tree_TestClass.checkErrorMessageWhenExecutingWithoutenteringCodeInImplementationInPythonPage(Tree_TestClass.java:392)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:269)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:76)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
]]>
    </failure>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInImplementationInPythonPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInImplementationInPythonPage" time="8.385">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInImplementationInPythonPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInImplementationInPythonPage" time="7.844">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInImplementationInPythonPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="ImplementationInPythonTryeditor" time="9.115"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="ImplementationInPythonTryeditor" time="9.473"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="ImplementationInPythonTryeditor" time="8.812"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="navigatebacktoImplementationInPython" time="8.595"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="BinaryTreeTraversals" time="7.452"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="BinaryTreeTraversalsTryhere" time="8.351"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInBinaryTreeTraversalsPage" time="8.459">
    <failure message="Test failed: Expected an error message when clicking the &#039;Run&#039; button without entering code, but no message was displayed" type="java.lang.AssertionError">
      <![CDATA[java.lang.AssertionError: Test failed: Expected an error message when clicking the 'Run' button without entering code, but no message was displayed
at org.testng.Assert.fail(Assert.java:111)
at dsAlgo_TestClasses.Tree_TestClass.checkErrorMessageWhenExecutingWithoutenteringCodeInBinaryTreeTraversalsPage(Tree_TestClass.java:438)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:269)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:76)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
]]>
    </failure>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInBinaryTreeTraversalsPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInBinaryTreeTraversalsPage" time="8.529">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInBinaryTreeTraversalsPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInBinaryTreeTraversalsPage" time="8.529">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInBinaryTreeTraversalsPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="BinaryTreeTraversalsTryeditor" time="9.553"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="BinaryTreeTraversalsTryeditor" time="8.624"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="BinaryTreeTraversalsTryeditor" time="8.728"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="navigatebacktoBinaryTreeTraversals" time="8.584"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="ImplementationOfBinaryTrees" time="7.620"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="ImplementationOfBinaryTreesTryhere" time="8.593"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInImplementationOfBinaryTreesPage" time="8.128">
    <failure message="Test failed: Expected an error message when clicking the &#039;Run&#039; button without entering code, but no message was displayed" type="java.lang.AssertionError">
      <![CDATA[java.lang.AssertionError: Test failed: Expected an error message when clicking the 'Run' button without entering code, but no message was displayed
at org.testng.Assert.fail(Assert.java:111)
at dsAlgo_TestClasses.Tree_TestClass.checkErrorMessageWhenExecutingWithoutenteringCodeInImplementationOfBinaryTreesPage(Tree_TestClass.java:484)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:269)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:76)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
]]>
    </failure>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInImplementationOfBinaryTreesPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInImplementationOfBinaryTreesPage" time="8.081">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInImplementationOfBinaryTreesPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInImplementationOfBinaryTreesPage" time="8.251">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInImplementationOfBinaryTreesPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="ImplementationOfBinaryTreesTryeditor" time="8.755"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="ImplementationOfBinaryTreesTryeditor" time="9.496"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="ImplementationOfBinaryTreesTryeditor" time="9.235"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="navigatebacktoImplementationOfBinaryTrees" time="8.501"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="ApplicationsOfBinaryTrees" time="7.031"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="ApplicationsOfBinaryTreesTryhere" time="8.221"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInApplicationsOfBinaryTreesPage" time="9.075">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInApplicationsOfBinaryTreesPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInApplicationsOfBinaryTreesPage" time="8.536">
    <failure message="Test failed: Expected an error message when clicking the &#039;Run&#039; button without entering code, but no message was displayed" type="java.lang.AssertionError">
      <![CDATA[java.lang.AssertionError: Test failed: Expected an error message when clicking the 'Run' button without entering code, but no message was displayed
at org.testng.Assert.fail(Assert.java:111)
at dsAlgo_TestClasses.Tree_TestClass.checkErrorMessageWhenExecutingWithoutenteringCodeInApplicationsOfBinaryTreesPage(Tree_TestClass.java:529)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:269)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:76)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
]]>
    </failure>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInApplicationsOfBinaryTreesPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInApplicationsOfBinaryTreesPage" time="8.523">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInApplicationsOfBinaryTreesPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="ApplicationsOfBinaryTreesTryeditor" time="8.793"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="ApplicationsOfBinaryTreesTryeditor" time="8.759"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="ApplicationsOfBinaryTreesTryeditor" time="8.529"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="navigatebacktoApplicationsOfBinaryTrees" time="9.310"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="BinarySearchTrees" time="8.260"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="BinarySearchTreesTryhere" time="8.280"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInBinarySearchTreesPage" time="8.676">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInBinarySearchTreesPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInBinarySearchTreesPage" time="8.546">
    <failure message="Test failed: Expected an error message when clicking the &#039;Run&#039; button without entering code, but no message was displayed" type="java.lang.AssertionError">
      <![CDATA[java.lang.AssertionError: Test failed: Expected an error message when clicking the 'Run' button without entering code, but no message was displayed
at org.testng.Assert.fail(Assert.java:111)
at dsAlgo_TestClasses.Tree_TestClass.checkErrorMessageWhenExecutingWithoutenteringCodeInBinarySearchTreesPage(Tree_TestClass.java:574)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:269)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:76)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
]]>
    </failure>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInBinarySearchTreesPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInBinarySearchTreesPage" time="8.831">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInBinarySearchTreesPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="BinarySearchTreesTryeditor" time="9.485"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="BinarySearchTreesTryeditor" time="9.545"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="BinarySearchTreesTryeditor" time="8.888"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="navigatebacktoBinarySearchTrees" time="8.414"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="ImplementationOfBST" time="7.222"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="ImplementationOfBSTTryhere" time="8.379"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInImplementationOfBSTPage" time="8.528">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInImplementationOfBSTPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInImplementationOfBSTPage" time="8.234">
    <skipped/>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInImplementationOfBSTPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="checkErrorMessageWhenExecutingWithoutenteringCodeInImplementationOfBSTPage" time="8.633">
    <failure message="Test failed: Expected an error message when clicking the &#039;Run&#039; button without entering code, but no message was displayed" type="java.lang.AssertionError">
      <![CDATA[java.lang.AssertionError: Test failed: Expected an error message when clicking the 'Run' button without entering code, but no message was displayed
at org.testng.Assert.fail(Assert.java:111)
at dsAlgo_TestClasses.Tree_TestClass.checkErrorMessageWhenExecutingWithoutenteringCodeInImplementationOfBSTPage(Tree_TestClass.java:619)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:269)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:76)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
]]>
    </failure>
  </testcase> <!-- checkErrorMessageWhenExecutingWithoutenteringCodeInImplementationOfBSTPage -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="ImplementationOfBSTTryeditor" time="9.377"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="ImplementationOfBSTTryeditor" time="8.908"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="ImplementationOfBSTTryeditor" time="9.018"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="navigatebacktoImplementationOfBST" time="7.999"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="PracticeQuestions" time="8.412"/>
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="PracticeQuestionspageisblank" time="13.557">
    <skipped/>
  </testcase> <!-- PracticeQuestionspageisblank -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="PracticeQuestionspageisblank" time="13.072">
    <error message="no such element: Unable to locate element: {&quot;method&quot;:&quot;xpath&quot;,&quot;selector&quot;:&quot;./div&quot;}
  (Session info: chrome=131.0.6778.204)
For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
Build info: version: &#039;4.29.0&#039;, revision: &#039;5fc1ec94cb&#039;
System info: os.name: &#039;Windows 11&#039;, os.arch: &#039;amd64&#039;, os.version: &#039;10.0&#039;, java.version: &#039;23.0.1&#039;
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [bedad2cfe29d9479c804c1b3b195ffa9, findElement {value=./div, using=xpath}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 131.0.6778.204, chrome: {chromedriverVersion: 131.0.6778.264 (2d05e315153..., userDataDir: C:\Users\mahaj\AppData\Loca...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:64570}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: windows, proxy: Proxy(), se:cdp: ws://localhost:64570/devtoo..., se:cdpVersion: 131.0.6778.204, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: bedad2cfe29d9479c804c1b3b195ffa9" type="org.openqa.selenium.NoSuchElementException">
      <![CDATA[org.openqa.selenium.NoSuchElementException: no such element: Unable to locate element: {"method":"xpath","selector":"./div"}
  (Session info: chrome=131.0.6778.204)
For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
Build info: version: '4.29.0', revision: '5fc1ec94cb'
System info: os.name: 'Windows 11', os.arch: 'amd64', os.version: '10.0', java.version: '23.0.1'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [bedad2cfe29d9479c804c1b3b195ffa9, findElement {value=./div, using=xpath}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 131.0.6778.204, chrome: {chromedriverVersion: 131.0.6778.264 (2d05e315153..., userDataDir: C:\Users\mahaj\AppData\Loca...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:64570}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: windows, proxy: Proxy(), se:cdp: ws://localhost:64570/devtoo..., se:cdpVersion: 131.0.6778.204, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: bedad2cfe29d9479c804c1b3b195ffa9
at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:501)
at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:485)
at org.openqa.selenium.remote.ErrorCodec.decode(ErrorCodec.java:167)
at org.openqa.selenium.remote.codec.w3c.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:138)
at org.openqa.selenium.remote.codec.w3c.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:50)
at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:215)
at org.openqa.selenium.remote.service.DriverCommandExecutor.invokeExecute(DriverCommandExecutor.java:216)
at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:174)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:545)
at org.openqa.selenium.remote.ElementLocation$ElementFinder$2.findElement(ElementLocation.java:165)
at org.openqa.selenium.remote.ElementLocation.findElement(ElementLocation.java:59)
at org.openqa.selenium.remote.RemoteWebDriver.findElement(RemoteWebDriver.java:368)
at org.openqa.selenium.remote.RemoteWebDriver.findElement(RemoteWebDriver.java:362)
at org.openqa.selenium.support.pagefactory.DefaultElementLocator.findElement(DefaultElementLocator.java:68)
at org.openqa.selenium.support.pagefactory.internal.LocatingElementHandler.invoke(LocatingElementHandler.java:38)
at jdk.proxy2/jdk.proxy2.$Proxy21.getText(Unknown Source)
at dsAlgo_PageFactory.Tree_PageFactory.childelements(Tree_PageFactory.java:200)
at dsAlgo_TestClasses.Tree_TestClass.PracticeQuestionspageisblank(Tree_TestClass.java:656)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:269)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:76)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
]]>
    </error>
  </testcase> <!-- PracticeQuestionspageisblank -->
  <testcase classname="dsAlgo_TestClasses.Tree_TestClass" name="PracticeQuestionspageisblank" time="12.967">
    <skipped/>
  </testcase> <!-- PracticeQuestionspageisblank -->
</testsuite> <!-- dsAlgo_TestClasses.Tree_TestClass -->
